test/classes/diten/multipledestructor.chpl

```diff
@@ -1,7 +1,7 @@
 class C {
-  proc ~C() { writeln("In ~C #1"); }
+  proc deinit() { writeln("In deinit #1"); }
 }
-proc C.~C() { writeln("In ~C #2"); }
+proc C.deinit() { writeln("In deinit #2"); }
```

Comment:
What you're describing is black-box testing (i.e., arguing from a position of having no knowledge of how the compiler works).  I assert (from a white-box viewpoint) that the implementation of deinit is so identical to deconstructors that if you broke deinit, you'd almost certainly break destructors as well, so would need to test everything that used a destructor (which is essentially _everything_) to be sure things are OK (and you'd need to do this to merge your code anyway).  Moreover, I think you'd be hard-pressed to break deinit in such a way that the deinit tests in test/initializers/ didn't break but this one did.  I also don't see how deinitializers are directly related to initializers.

I also don't think it's reasonable to assume that everything that uses init() and deinit() live under test/initializers.  As new features like these become available and the preferred way of doing things, they're going to crop up in arbitrary places (e.g., we didn't move binary-trees to test/initializers once it used init()).  Once deinit() is on master, I would hope none of us would write another destructor again (and to lazily convert ones we come across to deinit to prepare for the future).